pkg/controller/node/nodecontroller.go

```diff
@@ -112,6 +113,11 @@ type NodeController struct {
 	nodeStore      cache.StoreToNodeLister
 
 	forcefullyDeletePod func(*api.Pod)
+	availableCIDRs      sets.Int
+	// Record the first CIDR which has not been put into the availableCIDR set
+	firstAvailable int
```

Comment:
This var defines the first CIDR which has not been put into the availableCIDRs set. For example, we have totally 300 nodes. They need totally 300 CIDRs. Since the node controller will generate 256 new available CIDRs each time generateNewAvailabeCIDRs() being called, the function generateNewAvailabeCIDRs() will be called twice. Now the firstAvailable will be 512 which means next time when we want to generate new CIDRs and insert it into availableCIDRs set(call function generateNewAvailabeCIDRs()). Next time we we call generateNewAvailabeCIDRs(), it should begin from No. 512th CIDR which mean the firstAvailable=512.